kubernetes/template/DeploymentTemplates/script.sh

```diff
@@ -124,130 +150,114 @@ convert_to_cert() {
+log_level -i "MASTER_DNS_PREFIX:                $MASTER_DNS_PREFIX"
+log_level -i "MASTER_SIZE:                      $MASTER_SIZE"
+log_level -i "PUBLICIP_DNS:                     $PUBLICIP_DNS"
+log_level -i "PUBLICIP_FQDN:                    $PUBLICIP_FQDN"
+log_level -i "REGION_NAME:                      $REGION_NAME"
+log_level -i "RESOURCE_GROUP_NAME:              $RESOURCE_GROUP_NAME"
+log_level -i "SSH_PUBLICKEY:                    $SSH_PUBLICKEY"
+log_level -i "STORAGE_PROFILE:                  $STORAGE_PROFILE"
+log_level -i "TENANT_ID:                        $TENANT_ID"
+log_level -i "TENANT_SUBSCRIPTION_ID:           $TENANT_SUBSCRIPTION_ID"
 log_level -i "------------------------------------------------------------------------"
 
-#####################################################################################
-# Install all prequisite. 
-log_level -i "Update the system to latest."
-retrycmd_if_failure 5 10 sudo apt-get update -y
-
-log_level -i "Installing pax for string manipulation."
-retrycmd_if_failure 5 10 sudo apt-get install pax -y
-
-log_level -i "Installing jq for JSON manipulation."
-retrycmd_if_failure 5 10 sudo apt-get install jq -y
-
-log_level -i "Installing curl."
-retrycmd_if_failure 5 10 sudo apt-get install curl -y
-
-log_level -i "Installing apt-transport-https and lsb-release required for Azure CLI."
-retrycmd_if_failure 5 10 sudo apt-get install apt-transport-https lsb-release -y
-
-log_level -i "Installing software-properties-common and dirmngr required for Azure CLI."
-retrycmd_if_failure 5 10 sudo apt-get install software-properties-common dirmngr -y
-
-log_level -i "Update system again to latest."
-retrycmd_if_failure 5 10 sudo apt-get update -y
-
 ####################################################################################
-#Section to install Azure CLI.
-#https://docs.microsoft.com/en-us/azure/azure-stack/user/azure-stack-version-profiles-azurecli2#connect-to-azure-stack
-#https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-apt?view=azure-cli-latest 
+# Configure Azure CLI repository
+# https://docs.microsoft.com/en-us/azure/azure-stack/user/azure-stack-version-profiles-azurecli2#connect-to-azure-stack
+# https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-apt?view=azure-cli-latest
 
+# Add Azure CLI source
 AZ_REPO=$(lsb_release -cs)
-RECV_KEY=BC528686B50D79E339D3721CEB3E94ADBE1229CF
-
-# Modify your sources list
 echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $AZ_REPO main" | \
 sudo tee /etc/apt/sources.list.d/azure-cli.list
 
-#Get Microsoft signing key
+# Download Microsoft signing key
 log_level -i "Get the Microsoft signing key."
+RECV_KEY=BC528686B50D79E339D3721CEB3E94ADBE1229CF
 retrycmd_if_failure 5 10 sudo apt-key --keyring /etc/apt/trusted.gpg.d/Microsoft.gpg adv --keyserver packages.microsoft.com --recv-keys $RECV_KEY
 
-log_level -i "Update system again to latest."
-retrycmd_if_failure 5 10 sudo apt-get update 
```

Comment:
Also, i find doing apt-get update after package installations tend to fix any missing packages and version issues after the previous packages have been installed. If we are removing it, what guarantee do we have that the packages install have not put the system in a bad state?? Just a general comment

Reply:
I do not think `apt update` does that.
Quick link: https://linux.die.net/man/8/apt-get#update

Please share with me any resource supporting that claim. I'll like to learn